### ISSUE TITLE
Custom Word Separators Support

### ISSUE THREAD

**Issue by LenKiMo** [typora/typora-issues#1823]
For CJK languages, their symbols work as separators like `，` `。` `？` are fullwidth, which Typora can't identify them correctly when select text.

So if Advanced Config can allow user to appoint the custom word separators would be have good help to improve experience.

**Comment by abnerlee** [comment]
> which Typora can't identify them correctly when select text.

What does this mean? Which incorrect behavior will this cause?

**Comment by LenKiMo** [comment]
Well this is not a bug but a enhancement.

The expected effect is when double click a piece of text, it might be select the text from last symbol to next symbol, since CJK languages do not use space so much.

```
这是一段测试文本。用以表现CJK语言下文本选择的期望。
```

For example, when I need to select the first sentence or subsentence, double click the text, program might be select the whole piece `这是一段测试文本` from start until fullwidth comma `，`, or `用以表现CJK语言下文本选择的期望` from fullwidth comma to fullwidth full-stop. Since Typora does not recognize these fullwidth symbols, so it can't be select the whole sentence at once. If Typora support custom word separators, the select action can be enhanced for CJK users if they need.

But honestly I don't know if this is a pressing demand or just a small requirement. Since many of other software have mature participle which Typora also have, their first double click for a text is to select a word (like `这` `文本`) but not a whole sentence or subsentence, and the second is whole paragraph, which Typora act in same way. So I think it needs to be re-considered about if it is necessary,

**Comment by abnerlee** [comment]
From your description, this issue it not about separators like `，` ` 。` ` ？`, but incorrect bevahior of "select wrod" for Chinese text, can I understand like this?

**Comment by LenKiMo** [comment]
Yes, though as I thinked it can be achieved by separators. But I must need to admit that as for now, Typora's select word action work same as other software like Office Word in Chinese. So whether this Word Selection action needs to improved or not might need to re-consider.
